spec/policies/course_policy_spec.rb

```diff
@@ -25,4 +25,39 @@
       it { should_not permit(other_user, course) }
     end
   end
+
+  describe CoursePolicy::Scope do
+    let(:accredited_body) { create(:provider, :accredited_body, organisations: [organisation]) }
+    let(:training_provider) { create(:provider) }
+    let!(:course) { create(:course, provider: training_provider, accrediting_provider: accredited_body) }
+    let!(:other_course) { create(:course) }
+
+    subject { described_class.new(user, Course).resolve }
+
+    before { organisation }
+
+    context "user from the accredited_body" do
+      it { is_expected.to contain_exactly(course) }
+    end
+
+    context "user not from the accredited body" do
+      let(:organisation) { create(:organisation) }
+
+      it { is_expected.to be_empty }
+    end
+
+    context "a user from the provider" do
+      before do
+        accredited_body.organisations.clear
+        training_provider.organisations << organisation
+      end
```

Comment:
How about this instead of the pockery-jiggery with the organisations?

```
      let(:user) { training_provider.organisations.first.users.first }
```

Reply:
I've fixed this but couldn't quite do it the way you suggested. `training_provider.organisations.first.users` is empty so I added a new user and organisation in that context.